source/agent.js

```diff
@@ -440,6 +440,12 @@ class Agent extends EventEmitter {
 						session.destroy();
 					});
 
+					session.once('goaway', () => {
+						// Prevent session from being used for new requests.
+						// The session will eventually emit either an 'error' or 'close' event.
+						session[kGracefullyClosing] = true;
```

Comment:
Please use `gracefullyClose(session)` instead

Reply:
Can you add a comment to the spec as well please?

Reply:
@szmarczak My only concern with using `gracefullyClose(session)` is that `session.close()` may be fired multiple times but this condition appears to be handled [here](https://github.com/nodejs/node/blob/v16.x/lib/internal/http2/core.js#L1549) so I will update it to use `gracefullyClose`.

As for the spec, do you mean the [comment in the source code](https://github.com/nodejs/node/blob/v16.x/lib/internal/http2/core.js#L1166)?

Reply:
No, I mean the HTTP/2 RFC

Reply:
Does this work?
```js
// See https://datatracker.ietf.org/doc/html/rfc7540#section-6.8 

// There is an inherent race condition between an endpoint starting new streams and the remote sending a GOAWAY frame.
// Receivers of a GOAWAY frame MUST NOT open additional streams on the connection, although a new connection can be established for new streams.
```

Reply:
Change made to reference RFC